meilidb-http/src/routes/index.rs

```diff
@@ -12,104 +17,313 @@ use crate::models::token::ACL::*;
 use crate::routes::document::IndexUpdateResponse;
 use crate::Data;
 
+fn generate_uid() -> String {
+    let mut rng = rand::thread_rng();
+    let sample = b"abcdefghijklmnopqrstuvwxyz0123456789";
+    sample
+        .choose_multiple(&mut rng, 8)
+        .map(|c| *c as char)
+        .collect()
+}
+
 pub async fn list_indexes(ctx: Context<Data>) -> SResult<Response> {
     ctx.is_allowed(IndexesRead)?;
-    let list = ctx
-        .state()
-        .db
-        .indexes_names()
-        .map_err(ResponseError::internal)?;
-    Ok(tide::response::json(list))
+
+    let indexes_uids = ctx.state().db.indexes_uids();
+
+    let env = &ctx.state().db.env;
+    let reader = env.read_txn().map_err(ResponseError::internal)?;
+
+    let mut response_body = Vec::new();
+
+    for index_uid in indexes_uids {
+        let index = ctx.state().db.open_index(&index_uid);
+
+        match index {
+            Some(index) => {
+                let name = index
+                    .main
+                    .name(&reader)
+                    .map_err(ResponseError::internal)?
+                    .ok_or(ResponseError::internal("'name' not found"))?;
+                let created_at = index
+                    .main
+                    .created_at(&reader)
+                    .map_err(ResponseError::internal)?
+                    .ok_or(ResponseError::internal("'created_at' date not found"))?;
+                let updated_at = index
+                    .main
+                    .updated_at(&reader)
+                    .map_err(ResponseError::internal)?
+                    .ok_or(ResponseError::internal("'updated_at' date not found"))?;
+
+                let index_reponse = IndexResponse {
+                    name,
+                    uid: index_uid,
+                    created_at,
+                    updated_at,
+                };
+                response_body.push(index_reponse);
+            }
+            None => error!(
+                "Index {} is referenced on list of indexes but cannot be found",
+                index_uid
+            ),
+        }
+    }
+
+    Ok(tide::response::json(response_body))
 }
 
-pub async fn get_index_schema(ctx: Context<Data>) -> SResult<Response> {
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+struct IndexResponse {
+    name: String,
+    uid: String,
+    created_at: DateTime<Utc>,
+    updated_at: DateTime<Utc>,
+}
+
+pub async fn get_index(ctx: Context<Data>) -> SResult<Response> {
     ctx.is_allowed(IndexesRead)?;
 
     let index = ctx.index()?;
 
     let env = &ctx.state().db.env;
     let reader = env.read_txn().map_err(ResponseError::internal)?;
 
-    let schema = index
+    let uid = ctx.url_param("index")?.to_string();
```

Comment:
Do you really need to convert it into a string?
```suggestion
    let uid = ctx.url_param("index")?;
```